xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -1087,13 +1083,188 @@ void CActiveAE::Configure(AEAudioFormat *desiredFmt)
+      uint64_t avlayout = CAEUtil::GetAVChannelLayout(audioDSPOutputFormat.m_channelLayout);
+      audioDSPOutputFormat.m_channelLayout = CAEUtil::GetAEChannelLayout(avlayout);
+
+      //! @todo reimplement this with AudioDSP V2
+    //  if (stream)
+    //  {
+    //    if (!stream->m_inputBuffers)
+    //    {
+    //      // align input buffers with period of sink or encoder
+    //      stream->m_format.m_frames = audioDSPOutputFormat.m_frames * ((float)stream->m_format.m_sampleRate / audioDSPOutputFormat.m_sampleRate);
+
+    //      // create buffer pool
+    //      stream->m_inputBuffers = new CActiveAEBufferPool(stream->m_format);
+    //      stream->m_inputBuffers->Create(MAX_CACHE_LEVEL * 1000);
+    //      stream->m_streamSpace = stream->m_format.m_frameSize * stream->m_format.m_frames;
+
+    //      // if input format does not follow ffmpeg channel mask, we may need to remap channels
+    //      stream->InitRemapper();
+    //    }
+    //    if (stream->m_processingBuffers)
+    //    {
+    //      DSPErrorCode_t dspErr = m_audioDSP.ReleaseProcessingBuffer(stream->m_id);
+    //      if (dspErr != DSP_ERR_NO_ERR)
+    //      {//! @todo AudioDSP log AudioDSP error
+    //      }
+    //    }
+    //    if (!stream->m_processingBuffers)
+    //    {
+    //      stream->m_processingBuffers = m_audioDSP.GetProcessingBuffer(stream, audioDSPOutputFormat);//new CActiveAEStreamBuffers(stream->m_inputBuffers->m_format, audioDSPOutputFormat);
+    //      
+    //      // set stream infos
+    //      //! @todo AudioDSP reimplement this
+    //      //stream->m_processingBuffers->SetExtraData(stream->m_profile, stream->m_matrixEncoding, stream->m_audioServiceType);
+
+    //      // set resampler options
+    //      //! @todo AudioDSP reimplement this
+    //      //stream->m_processingBuffers->ConfigureResampler(m_settings.normalizelevels, m_settings.stereoupmix, m_settings.resampleQuality);
+    //      //stream->m_processingBuffers->ForceResampler(stream->m_forceResampler);
+    //    }
+
+    //    if (!stream->m_processingBuffers)
+    //    {//! @todo AudioDSP error handling for failed buffer creation
+    //      CLog::Log(LOGERROR, "ActiveAE::%s - failed to create AudioDSP processing buffer!", __FUNCTION__);
+    //      m_stats.SetSinkCacheTotal(0);
+    //      m_stats.SetSinkLatency(0);
+    //      AEAudioFormat invalidFormat;
+    //      invalidFormat.m_dataFormat = AE_FMT_INVALID;
+    //      m_stats.SetCurrentSinkFormat(invalidFormat);
+    //      m_extError = true;
+    //      return;
+    //    }
+    //    stream->m_processingBuffers->Create(MAX_CACHE_LEVEL * 1000);
+    //  }
+
+    //  //! @todo adjust to decoder
+    //  audioDSPOutputFormat = stream->m_processingBuffers->m_outputFormat;
+    //  m_sinkRequestFormat = audioDSPOutputFormat;
+    }
+
+    std::list<CActiveAEStream*>::iterator it;
+    for (it = m_streams.begin(); it != m_streams.end(); ++it)
```

Comment:
Also for things like them is mostly now only the `auto` requested (but for me is sometimes the old better to see what it is ;-) )